src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Pages/WeatherDetails.razor

```diff
@@ -0,0 +1,37 @@
+@page "/weather/details/{id:int}"
+@*#if (!InteractiveAtRoot) -->
+@attribute [StreamRendering]
+##endif*@
+@inject NavigationManager NavigationManager
+
+@if (weatherDetails == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    <div>
+        <p><strong>Date:</strong> @weatherDetails.Date.ToShortDateString()</p>
+        <p><strong>Temperature (C):</strong> @weatherDetails.TemperatureC</p>
+        <p><strong>Temperature (F):</strong> @weatherDetails.TemperatureF</p>
+        <p><strong>Summary:</strong> @weatherDetails.Summary</p>
+    </div>
+}
+
+@code{
+    [Parameter]
+    public int Id { get; set; }
+
+    private WeatherForecast? weatherDetails;
+
+    protected override async Task OnInitializedAsync()
+    {
+        weatherDetails = null;
+
+        // Simulate fetching data from a database
+        await Task.Delay(500);
```

Comment:
This delay makes `NotFound` throw in SSR streaming scenarios. It does not look good on a template.

Reply:
We can switch off streaming but then, `Router.OnNotFound` is triggered after `EndpointHtmlRenderer.SetNotFoundResponse`, so renderer is already stopped -> we don't see result of calling `NavigationManager.NotFound`.

Reply:
Decision: template changes will be restricted to adding `NotFound` page.